lib/sycamore/sycamore/transforms/llm_query.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from sycamore.data import Element, Document
+from sycamore.llms.openai import OpenAIModels
+from sycamore.plan_nodes import NonCPUUser, NonGPUUser, Node
+from sycamore.llms import OpenAI, OpenAIClientWrapper
+from sycamore.transforms.map import Map
+from sycamore.utils.time_trace import timetrace
+
+
+class LLMQueryAgent(ABC):
```

Comment:
What's the rationale for this interface? Do we expect multiple implementations?

Reply:
Discussed offline, can add it in later if we add more implementations. Have removed it for now